### PR TITLE
ci: fix test splits that have less test packages than runner count from hanging

### PR DESCRIPTION
**Pull request by jmurret**
### Description
In 1.13, there are only two compatibility test packages.  So when a generate matrix job has a TOTAL_RUNNERS of 5 but there are only 2 packages, the process hangs.  This PR changes the runner count to 2 in this case.  

Another issue that occurs that is fixed in this PR is that when you specify TOTAL_RUNNERS = 4, you will actually split across 5 runners.  (The same occurs with any number you set TOTAL_RUNNERS to.  You will actually get 1 greater.)

### CODE REVIEW

File: .github/workflows/test-integrations.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 name: test-integrations
 
 on:
+  push:
```
Comment (loshz) [inline .github/workflows/test-integrations.yml:7]: Was this added for debugging?

File: .github/workflows/test-integrations.yml [hunk 2]
```diff
@@ -155,17 +156,25 @@ jobs:
...
         run: |
+          NUM_RUNNERS=$TOTAL_RUNNERS
+          NUM_DIRS=$(find ./test/integration/connect/envoy -maxdepth 1 -type d | wc -l)
+          if [ $NUM_DIRS -lt $NUM_RUNNERS ]
+          then
+            echo "TOTAL_RUNNERS is larger than the number of tests/packages to split."
+            NUM_RUNNERS=$NUM_DIRS
+          fi
+          # fix issue where test splitting calculation generates 1 more split than TOTAL_RUNNERS.
+          NUM_RUNNERS=$(($NUM_RUNNERS-1))
```
Comment (loshz) [inline .github/workflows/test-integrations.yml:170]:
```suggestion
          if [ "$NUM_DIRS" -lt "$NUM_RUNNERS" ]; then
            echo "TOTAL_RUNNERS is larger than the number of tests/packages to split."
            NUM_RUNNERS=$NUM_DIRS
          fi
          # fix issue where test splitting calculation generates 1 more split than TOTAL_RUNNERS.
          NUM_RUNNERS=$((NUM_RUNNERS-1))
```

File: .github/workflows/test-integrations.yml [hunk 3]
```diff
@@ -261,18 +270,26 @@ jobs:
...
           cd ./test/integration/consul-container
+          NUM_RUNNERS=$TOTAL_RUNNERS
+          NUM_DIRS=$(find ./test -maxdepth 2 -type d | wc -l)
+          if [ $NUM_DIRS -lt $NUM_RUNNERS ]
+          then
+            echo "TOTAL_RUNNERS is larger than the number of tests/packages to split."
+            NUM_RUNNERS=$NUM_DIRS
+          fi
+          # fix issue where test splitting calculation generates 1 more split than TOTAL_RUNNERS.
+          NUM_RUNNERS=$(($NUM_RUNNERS-1))
```
Comment (loshz) [inline .github/workflows/test-integrations.yml:285]:
```suggestion
          if [ "$NUM_DIRS" -lt "$NUM_RUNNERS" ]; then
            echo "TOTAL_RUNNERS is larger than the number of tests/packages to split."
            NUM_RUNNERS=$NUM_DIRS
          fi
          # fix issue where test splitting calculation generates 1 more split than TOTAL_RUNNERS.
          NUM_RUNNERS=$((NUM_RUNNERS-1))
```

File: .github/workflows/test-integrations.yml [hunk 4]
```diff
@@ -359,17 +376,25 @@ jobs:
...
           cd ./test/integration/consul-container/test/upgrade
+          NUM_RUNNERS=$TOTAL_RUNNERS
+          NUM_DIRS=$(go test ./...  -list=. -json | jq -r '.Output | select (. !=null) |  select(. | startswith("Test")) | gsub("[\\n\\t]"; "")' | wc -l)
+          if [ $NUM_DIRS -lt $NUM_RUNNERS ]
+          then
+            echo "TOTAL_RUNNERS is larger than the number of tests/packages to split."
+            NUM_RUNNERS=$NUM_DIRS
+          fi
+          # fix issue where test splitting calculation generates 1 more split than TOTAL_RUNNERS.
+          NUM_RUNNERS=$(($NUM_RUNNERS-1))
```
Comment (loshz) [inline .github/workflows/test-integrations.yml:391]:
```suggestion
          if [ "$NUM_DIRS" -lt "$NUM_RUNNERS" ]; then
            echo "TOTAL_RUNNERS is larger than the number of tests/packages to split."
            NUM_RUNNERS=$NUM_DIRS
          fi
          # fix issue where test splitting calculation generates 1 more split than TOTAL_RUNNERS.
          NUM_RUNNERS=$((NUM_RUNNERS-1))
```

**Review by loshz — COMMENTED** [review]
Just the one comment about the `push` trigger. The rest are non-blocking syntactic bash NITs.

File: .github/workflows/test-integrations.yml (hunk 1 above)
Comment (jmurret) [reply .github/workflows/test-integrations.yml]: Fixed.  ty!

**Review by loshz — APPROVED** [review]

**Pull request merged**